tracer/test/Datadog.Trace.IntegrationTests/StatsTests.cs

```diff
@@ -440,6 +524,25 @@ void WaitForStats(AutoResetEvent e, bool expected)
                 }
             }
 
+            void AssertTraces(IReadOnlyList<MockSpan> payload, bool expectStats)
+            {
+                // All the spans generate the same stats point, so only the first
+                // occurrence would be considered "rare". However, the RareSampler
+                // only runs after the PrioritySampler, so assert the metric "_dd.rare"
+                // based on the SamplingPriority
+                bool expectRareMetric = expectStats && payload[0].GetMetric(Metrics.SamplingPriority) <= 0;
```

Comment:
As I understand it, the rare metric runs _first_, so I think this comment is incorrect? 🤔 I think we always expect payload[0] to have it regardless of sampling priority?